docs/user/troubleshooting.md

```diff
@@ -99,6 +99,230 @@ This is safe to ignore and merely indicates that the etcd bootstrapping is still
+network                                         True        False         False     13m
+node-tuning                                     True        False         False     11m
+openshift-apiserver                             True        False         False     15m
+openshift-authentication                        True        False         False     20m
+openshift-cloud-credential-operator             True        False         False     18m
+openshift-controller-manager                    True        False         False     10m
+openshift-samples                               True        False         False     8m42s
+operator-lifecycle-manager                      True        False         False     17m
+service-ca                                      True        False         False     30m
+
+$ oc --config=${INSTALL_DIR}/auth/kubeconfig get clusteroperator monitoring -oyaml
+apiVersion: config.openshift.io/v1
+kind: ClusterOperator
+metadata:
+  creationTimestamp: 2019-02-27T22:47:04Z
+  generation: 1
+  name: monitoring
+  resourceVersion: "24677"
+  selfLink: /apis/config.openshift.io/v1/clusteroperators/monitoring
+  uid: 9a6a5ef9-3ae1-11e9-bad4-0a97b6ba9358
+spec: {}
+status:
+  conditions:
+  - lastTransitionTime: 2019-02-27T22:49:10Z
+    message: Successfully rolled out the stack.
+    status: "True"
+    type: Available
+  - lastTransitionTime: 2019-02-27T22:49:10Z
+    status: "False"
+    type: Progressing
+  - lastTransitionTime: 2019-02-27T22:49:10Z
+    status: "False"
+    type: Failing
+  extension: null
+  relatedObjects: null
+  version: ""
+```
+
+Again, the cluster operators also publish [conditions][cluster-operator-conditions] like `Failing`, `Available` and `Progressing` that can help user provide information on the current state of the operator:
+
+```console
+$ oc--config=${INSTALL_DIR}/auth/kubeconfig get clusteroperator monitoring -o=jsonpath='{.status.conditions}'
+[map[lastTransitionTime:2019-02-27T22:49:10Z message:Successfully rolled out the stack. status:True type:Available] map[lastTransitionTime:2019-02-27T22:49:10Z status:False type:Progressing] map[status:False type:Failing lastTransitionTime:2019-02-27T22:49:10Z]]
+```
+
+Each clusteroperator also publishes the list of objects owned by the cluster operator. To get the information:
+
+```console
+$ oc --config=${INSTALL_DIR}/auth/kubeconfig get clusteroperator kube-apiserver -o=jsonpath='{.status.relatedObjects}'
+[map[resource:kubeapiservers group:operator.openshift.io name:cluster] map[group: name:openshift-config resource:namespaces] map[group: name:openshift-config-managed resource:namespaces] map[group: name:openshift-kube-apiserver-operator resource:namespaces] map[group: name:openshift-kube-apiserver resource:namespaces]]
+```
+
+Failing to initialize the cluster is usually not a fatal failure in terms of cluster creation as the user can look at the failures from `clusteroperator` which will unblock `cluster-version-operator` to make progress.
```

Comment:
While this is true, I don't know that it's worth calling it out here.  Ideally, we're promoting release images that Just Work, and this troubleshooting doc is really about "how do I gather enough information to make a bug report?" or "am I just holding this wrong?".  The things we can configure in the installer should be limited enough that folks can collect the bug-report info, and then destroy the cluster and start over.  I don't want to talk folks through recovering broken clusters.

Reply:
from https://github.com/openshift/installer/blob/master/docs/user/troubleshooting.md#installer-troubleshooting
```
In these events, it is helpful to understand the likely failure modes as well as how to troubleshoot the failure.
```

this doc is meant to be guide on understanding the failures and just a way to write bug reports. And I tried to make a note that `failed to initialize` is not a fatal failure that would require destroy and create. The user should take the state of `clusterversion` and `clusteroperators` to decide that.

Reply:
> this doc is meant to be guide on understanding the failures and *NOT* just a way to write bug reports.

I don't see a lot of space between understanding failures and being able to write good bug reports.  *Fixing* things seems like a bigger leap.  But if you don't expect this to lead to extra support cost for our team, that's enough for me.